File: framework/web/src/test/java/cn/bestwu/simpleframework/web/serializer/HideSerializerTest.java
package cn.bestwu.simpleframework.web.serializer;

import cn.bestwu.lang.util.StringUtil;
import cn.bestwu.simpleframework.web.serializer.annotation.JsonHide;
import com.fasterxml.jackson.core.JsonProcessingException;
import com.fasterxml.jackson.databind.ObjectMapper;
import org.junit.Test;

/**
 * @author Peter Wu
 */
public class HideSerializerTest {

  class User {

    @JsonHide(beginKeep = 2, endKeep = 2)
    String tel;
    @JsonHide(beginKeep = 1, endKeep = 1)
    String password;

    public String getTel() {
      return tel;
    }

    public void setTel(String tel) {
      this.tel = tel;
    }

    public String getPassword() {
      return password;
    }

    public void setPassword(String password) {
      this.password = password;
    }
  }

  ObjectMapper objectMapper = new ObjectMapper();

  @Test
  public void test() throws JsonProcessingException {
    User user = new User();
    user.setPassword("1");
    user.setTel("18000000000");
    System.err.println(StringUtil.valueOf(objectMapper.writeValueAsString(user)));

  }
}

File: web/src/test/kotlin/top/bettercode/summer/test/web/serializer/HideSerializerTest.kt
package top.bettercode.summer.test.web.serializer

import org.junit.jupiter.api.Test
import top.bettercode.summer.tools.lang.util.StringUtil.json
import top.bettercode.summer.web.serializer.annotation.JsonHide

/**
 * @author Peter Wu
 */
class HideSerializerTest {
    internal class User {
        @JsonHide(beginKeep = 2, endKeep = 2)
        var tel: String? = null

        @JsonHide(beginKeep = 1, endKeep = 1)
        var password: String? = null
    }

    @Test
    fun test() {
        val user = User()
        user.password = "1"
        user.tel = "18000000000"
        System.err.println(json(user))
    }
}